core/backend/src/SqliteChangesetReader.ts

```diff
@@ -0,0 +1,279 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { IModelJsNative } from "@bentley/imodeljs-native";
+import { DbOpcode, DbResult, IDisposable } from "@itwin/core-bentley";
+import { ECDb } from "./ECDb";
+import { IModelDb } from "./IModelDb";
+import { IModelHost } from "./IModelHost";
+
+/** Changed value type
+ * @beta
+*/
+type SqliteValue = Uint8Array | number | string | null | undefined;
+
+/**  Array of changed values
+ * @beta
+*/
+type SqliteValueArray = SqliteValue[];
+export interface ChangeFormatArgs {
+  includeTableName?: true;
+  includeOpCode?: true;
+  includeNullColumns?: true;
+  includeStage?: true;
+  includePrimaryKeyInUpdateNew?: true;
+}
+
+/** Operation that cause the change
+ * @beta
+*/
+export type SqliteChangeOp = "Inserted" | "Updated" | "Deleted";
+
+/** Stage is version of value that needed to be read
+ * @beta
+*/
+export type SqliteValueStage = "Old" | "New";
+
+/** Db from which schema will be read. It should be from timeline to which changeset belong.
+ * @beta
+*/
+export type AnyDb = IModelDb | ECDb;
+
+/** Arg to open a changeset file from disk
+ * @beta
+*/
+export interface SqliteChangesetReaderArgs {
+  readonly db?: AnyDb; /** db from which schema will be read. It should be close to changeset.*/
+  readonly invert?: true; /** invert the changeset operations */
+  readonly disableSchemaCheck?: true; /** do not check if column of change match db schema instead ignore addition columns */
+}
+
+/**
+ * @beta
+ */
+export interface SqliteChange {
+  $table?: string;
```

Comment:
What motivates prepending `$` to these property names?

Reply:
Do you expect `[key: string]: any` to contain `op`, `table`, and/or `stage`?

Reply:
to avoid conflict with column names.